lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -30,48 +31,69 @@ const ignoredPathComponents = [
   '**/*-renderer.js',
 ];
 
-// @ts-ignore - @types/esprima lacks all of these
-function computeDescription(ast, property, value, startRange) {
-  const endRange = property.range[0];
-
-  for (const comment of ast.comments || []) {
-    if (comment.range[0] < startRange) continue;
-    if (comment.range[0] > endRange) continue;
-    if (comment.value.includes('@')) {
-      // This is a complex description with description and examples.
-      let description = '';
-      /** @type {Record<string, string>} */
-      const examples = {};
-
-      const r = /@(\w+) ({.+})?(.*)(\n|$)/g;
-      let matches;
-      while ((matches = r.exec(comment.value)) !== null) {
-        const tagName = matches[1];
-        const example = matches[2];
-        const messageOrPlaceholder = matches[3].trim();
-
-        if (tagName === 'description') {
-          description = messageOrPlaceholder;
-        } else if (tagName === 'example') {
-          examples[messageOrPlaceholder] = example.substring(1, example.length - 1);
-        }
-      }
+/**
+ * Extract the description and examples (if any) from a jsDoc annotation.
+ * @param {import('typescript').JSDoc|undefined} ast
+ * @param {string} message
+ * @return {{description: string, examples: Record<string, string>}}
+ */
+function computeDescription(ast, message) {
+  if (!ast) {
+    throw Error(`Missing description comment for message "${message}"`);
+  }
 
-      // Make sure description is not empty
-      if (description.length === 0) throw Error(`Empty @description for message "${value}"`);
-      return {description, examples};
+  if (ast.tags) {
+    // This is a complex description with description and examples.
+    let description = '';
+    /** @type {Record<string, string>} */
+    const examples = {};
+
+    for (const tag of ast.tags) {
+      const comment = coerceToSingleLineAndTrim(tag.comment);
+
+      if (tag.tagName.text === 'description') {
+        description = comment;
+      } else if (tag.tagName.text === 'example') {
+        const {placeholderName, exampleValue} = parseExampleJsDoc(comment);
+        examples[placeholderName] = exampleValue;
+      } else {
+        throw new Error('bad tag' + tag.tagName.text);
```

Comment:
```suggestion
        throw new Error(`Unexpected tagName "@${tag.tagName.text}"`);
```

Reply:
Haha, whoops, I threw that in there to see how hard it was to accidentally trigger that case with reasonable typos and then I guess accidentally committed it. Maybe we'll go with throwing in that case after all :)